drivers/timer/stm32_lptim_timer.c

```diff
@@ -226,6 +235,16 @@ void z_clock_set_timeout(s32_t ticks, bool idle)
 	if (next_arr > LPTIM_TIMEBASE) {
 		next_arr = LPTIM_TIMEBASE;
 	}
+
+	/* If we are close to the roll over of the ticker counter
+	 * change current tick so it can be compared with buffer.
+	 * If this event got outdated fire interrupt right now,
+	 * else schedule it normally.
+	 */
+	if (next_arr <= ((LL_LPTIM_GetCounter(LPTIM1) + 1) & LPTIM_TIMEBASE)) {
+		NVIC_SetPendingIRQ(LPTIM1_IRQn);
+		lptim_fired = 1;
+	}
```

Comment:
LPTIM1 current value must be read like this:
```
	while (lp_time != LL_LPTIM_GetCounter(LPTIM1)) {
		lp_time = LL_LPTIM_GetCounter(LPTIM1);
	}
```

Reply:
changed LL_LPTIM_GetCounter  with lp_time value as this variable is already set.